tests/domains/user/service/user.service.impl.test.ts

```diff
@@ -0,0 +1,45 @@
+import { before, beforeEach, describe, it, mock } from 'node:test';
+import assert from 'node:assert';
+import { NotFoundException } from '@utils';
+import { UserServiceImpl, type UserService } from '@domains/user/service';
+import { UserDTO, type UserRepository } from '@domains/user';
+import { UserMockRepository } from '../mockRepository/userMockRepository';
+
+let mockRepository: UserRepository;
+let service: UserService;
+let user: UserDTO;
+
+describe('user getById tests', () => {
+  before(() => {
+    mockRepository = new UserMockRepository();
+    service = new UserServiceImpl(mockRepository);
+    user = new UserDTO({
+      id: 'id',
+      profileImage: null,
+      projectsRoleAssigned: [],
+      createdAt: new Date('2023-11-18T19:28:40.065Z'),
+      deletedAt: null,
+      emittedUserProjectRole: [],
+    });
+  });
+
+  beforeEach(() => {
+    mock.restoreAll();
+  });
+
+  it('Should successfully get a user by a provided id', async () => {
+    const recievedUser: UserDTO = await service.getById('id');
+
+    assert.notEqual(recievedUser, null);
```

Comment:
Redundant validation. If user is null, id won't be equal to the provided one

Reply:
Fixed